pkg/operator/configobserver/cloudconfig/sync_cloud_config.go

```diff
@@ -0,0 +1,121 @@
+package cloudprovider
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	corelisterv1 "k8s.io/client-go/listers/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+const (
+	configNamespace                 = "openshift-config"
+	machineSpecifiedConfigNamespace = "openshift-config-managed"
+	machineSpecifiedConfig          = "kube-cloud-config"
+)
+
+var validCloudProviders = sets.NewString("aws", "azure", "gce", "openstack", "vsphere")
+
+// InfrastructureLister lists infrastrucre information and allows resources to be synced
+type InfrastructureLister interface {
+	InfrastructureLister() configlistersv1.InfrastructureLister
+	ResourceSyncer() resourcesynccontroller.ResourceSyncer
+	ConfigMapLister() corelisterv1.ConfigMapLister
+}
+
+// NewCloudConfigSyncer returns a new cloudprovider dependent observer for syncing cloud config per provider
+// for controller-manager and api-server.
+func NewCloudConfigSyncer(targetNamespaceName string) configobserver.ObserveConfigFunc {
```

Comment:
who will consume this?

This package here is about config observers. A controller is misplaced.

Reply:
This is still a blocker for this PR.